File: binary search/valid_perfect_square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1: 
            return True
        
        l = 1
        h = num
        while l <= h:
            mid = l + int((h - l) / 2)
            if mid*mid == num:
                return True
            elif mid*mid > num : h = mid-1
            elif mid*mid < num : l = mid + 1
        return False

File: binary search/valid_perfect_square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        
        if(num == 1)
            return true;
        long l = 1;
        long h = num;
        
        while(l<=h)
        {
            long mid = l + (h-l)/2;
            if (mid*mid == num)
                return true;
            
            else if (mid*mid > num)
                h = mid-1;
            
            else
                l = mid+1;
        }
        
        return false;
    }
}